ssh/tailssh/tailssh.go

```diff
@@ -96,16 +96,17 @@ type server struct {
 
 	// mu protects activeSessions.
 	mu             sync.Mutex
-	activeSessions map[string]bool
+	activeSessions map[string]*tailcfg.SSHAction
 }
 
 var debugPolicyFile = envknob.String("TS_DEBUG_SSH_POLICY_FILE")
 
-// portForward reports whether the ctx should be allowed to port forward
+// localPortForward reports whether the ctx should be allowed to port forward
 // to the specified host and port.
 // TODO(bradfitz/maisem): should we have more checks on host/port?
-func (srv *server) portForward(ctx ssh.Context, destinationHost string, destinationPort uint32) bool {
-	return srv.isActiveSession(ctx)
+func (srv *server) localPortForward(ctx ssh.Context, destinationHost string, destinationPort uint32) bool {
+	a := srv.sessionAction(ctx)
+	return a.LocalPortForwarding
```

Comment:
I was going to say your API change on line 313 below was prone to misuse and this here looks what I was afraid of: nil ptr derefs after blind use trusting it'll return the right thing.

Reply:
Sigh, yes. Fixed